qml/CoverFileItem.qml

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2013-2014 Jolla Ltd.
+ * Contact: Robin Burchell <robin.burchell@jolla.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+Item {
+    id: root
+
+    property alias text: label.text
+    property bool multiLine
+    property string iconSource
+
+    Image {
+        id: icon
+
+        anchors {
+            left: parent.left
+            leftMargin: Theme.paddingLarge
+            verticalCenter: root.multiLine ? undefined : parent.verticalCenter
+        }
+        source: root.iconSource !== "" ? root.iconSource
+                                       : "image://theme/icon-l-document"
```

Comment:
Yes, the size -l- issue is visible only in the main page, not in cover since cover resize icons (notice this after). The comment was position on the first -l- I've found, so quite misplaced !

Not having a post-script icon is not much of an issue currently since post-script are not handled anyway… Mentioning them was just to give a use case for when the icon-l-document is actually shown.

For the scalable icons, one should find the SVG sources of the current PNG files and put them in the images/ folder. Then generate whatever sizes requested by scalability and use the new SAILFISHAPP_ICONS variable in the pro file. Or is there a possibility to add images to an icon theme ? What is the best way to go ? Anyway, first one need the SVG sources…